pkg/distributor/ha_tracker.go

```diff
@@ -513,7 +513,8 @@ func (e replicasNotMatchError) Error() string {
 	return fmt.Sprintf("replicas did not mach, rejecting sample: replica=%s, elected=%s", e.replica, e.elected)
 }
 
-// Needed for errors.Is to work properly.
+// Is implements the interface expected by errors.Is.
+//nolint:errorlint
```

Comment:
I'm curious: what is the complain of the linter here?

Reply:
Wants to use `errors.As()` instead of `err.(replicasNotMatchError)`

Reply:
I'd create an issue for this: we're tweaking the of `errors.Is` to behave as `errors.As`.